src/eureka/S5_lightcurve_fitting/models/BatmanModels.py

```diff
@@ -288,42 +352,48 @@ def eval(self, channel=None, **kwargs):
                 time = split([time, ], self.nints, chan)[0]
 
-            # Set all parameters
-            for index, item in enumerate(self.longparamlist[chan]):
-                setattr(bm_params, self.paramtitles[index],
-                        self.parameters.dict[item][0])
-
-            # Enforce physicality to avoid crashes from batman by
-            # returning something that should be a horrible fit
-            if not ((0 < bm_params.per) and (0 < bm_params.inc < 90) and
-                    (1 < bm_params.a) and (0 <= bm_params.ecc < 1)):
-                # Returning nans or infs breaks the fits, so this was
-                # the best I could think of
-                lcfinal = np.append(lcfinal, 1e12*np.ones_like(time))
-                continue
-
-            bm_params.limb_dark = 'uniform'
-            bm_params.u = []
-
-            if self.compute_ltt:
-                self.adjusted_time = correct_light_travel_time(time, bm_params)
-            else:
-                self.adjusted_time = time
+            light_curve = np.ones_like(time)
+            for pid in range(self.num_planets):
+                # Initialize planet
+                bm_params = PlanetParams(self, pid)
 
-            if not np.any(['t_secondary' in key
-                           for key in self.longparamlist[chan]]):
-                # If not explicitly fitting for the time of eclipse, get
-                # the time of eclipse from the time of transit, period,
-                # eccentricity, and argument of periastron
-                m_transit = batman.TransitModel(bm_params, self.adjusted_time,
-                                                transittype='primary')
-                bm_params.t_secondary = m_transit.get_t_secondary(bm_params)
+                # Set limb darkening parameters
+                bm_params.u = []
+                bm_params.limb_dark = 'uniform'
 
-            # Make the eclipse model
-            m_eclipse = batman.TransitModel(bm_params, self.adjusted_time,
-                                            transittype='secondary')
+                # Enforce physicality to avoid crashes from batman by
+                # returning something that should be a horrible fit
+                if not ((0 < bm_params.per) and (0 < bm_params.inc < 90) and
+                        (1 < bm_params.a) and (0 <= bm_params.ecc < 1)):
+                    # Returning nans or infs breaks the fits, so this was
+                    # the best I could think of
+                    light_curve = 1e12*np.ones_like(time)
+                    continue
 
-            lcfinal = np.append(lcfinal, m_eclipse.light_curve(bm_params))
+                # Compute light travel time
+                if self.compute_ltt:
+                    self.adjusted_time.append(
+                        correct_light_travel_time(time, bm_params))
+                else:
+                    self.adjusted_time.append(time)
```

Comment:
I have made edits so that it matches what's written in `PoetModels.py`.  The idea is to compute the adjusted time only once per planet and reuse the times for each spectroscopic channel.